R/ingest-scp.R

```diff
@@ -139,45 +122,28 @@ get_incidence_data <- function(state, chosen_cancer, chosen_race, chosen_sex, ch
 #'    "wa",
 #'    current_row
 #' )
-#'
 
 process_row <- function(state, row) {
-  current_cancer <- as.character(row["cancer_type"])
-  current_race <- as.character(row["race"])
-  current_sex <- as.character(row["sex"])
-  current_age <- as.character(row["age"])
-  current_stage <- as.character(row["stage"])
-  current_timespan <- as.character(row["timespan"])
+  row <- map_chr(row, as.character)
 
-  print(
-    paste(
-      "Processing:",
-      current_cancer, current_race, current_sex,
-      current_age, current_stage, current_timespan
-    )
-  )
-
-  current_df <- tryCatch({
+  tryCatch({
     get_incidence_data(
       state,
-      current_cancer,
-      current_race,
-      current_sex,
-      current_age,
-      current_stage,
-      current_timespan
+      row["cancer_type"],
+      row["race"],
+      row["sex"],
+      row["age"],
+      row["stage"],
+      row["timespan"]
     )
   }, error = function(e) {
-    # More detailed error message
-    message(paste("Error in row with cancer type:", current_cancer,
-                  "race:", current_race, "sex:", current_sex,
-                  "age:", current_age, "stage:", current_stage,
-                  "timespan:", current_timespan,
-                  " - Error message:", e$message))
-    return(NULL)  # Return NULL in case of error
+    paste0(c(row["cancer_type"],
+           row["race"],
+           row["sex"],
+           row["age"],
+           row["stage"],
+           row["timespan"]), collapse = ",")
   })
-
-  return(current_df)
 }
```

Comment:
This is a little complex but it can help us in future debugging. Let's talk about this synchronously.